src/aws/auth.rs

```diff
@@ -11,6 +11,22 @@ use vector_config::configurable_component;
 // default rather than relying on the SDK default to not change
 const DEFAULT_LOAD_TIMEOUT: Duration = Duration::from_secs(5);
 
+/// IMDS Client Configuration for authenticating with AWS.
+#[configurable_component]
+#[derive(Copy, Clone, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields)]
+pub struct ImdsAuthentication {
+    /// Number of IMDS retries for fetching tokens & metadata
+    max_attempts: u32,
+
+    /// Connect timeout for IMDS, in seconds.
+    connect_timeout: u64,
+
+    /// Read timeout for IMDS, in seconds.
+    read_timeout: u64,
```

Comment:
Could the two of these be updated to be `Duration` instead of `u64`?

It should be able to pulled from something like [`reduce`](https://github.com/vectordotdev/vector/blob/master/src/transforms/reduce/mod.rs#L40-L45), this would also make the "in seconds" portion of the comment remove-able since it'll be generated by the macro.